corehq/apps/receiverwrapper/tests/data/form_with_vault_item.xml

```diff
@@ -0,0 +1,20 @@
+<?xml version='1.0' ?>
+<form name="v2 create" version="1" uiVersion="1" xmlns:jrm="http://openrosa.org/jr/xforms" xmlns="http://commcarehq.org/test/submit">
+    <meta>
+        <DeviceID>Chalimbana</DeviceID>
+        <TimeStart>2010-06-29T13:42:50.137</TimeStart>
+        <TimeEnd>2010-06-29T13:55:50.137</TimeEnd>
+        <username>someuser</username>
+        <userID>ad38211be256653bceac8e2156475665</userID>
+        <instanceID>ad38211be256653bceac8e2156475666</instanceID>
+    </meta>
+	<case xmlns="http://commcarehq.org/case/transaction/v2" case_id="ad38211be256653bceac8e2156475667" user_id="ad38211be256653bceac8e2156475665" date_modified="2010-06-29T13:42:50" >
+	    <create>
+	        <secret_case_property>0123456789</secret_case_property>
```

Comment:
normally a form would have this as part of the form data as well as in a case property. If you have this in both places do you end up with 2 vault entries or just one? (Looks like it will only create one but seems like something worth having a test case for).

Reply:
Not sure what you mean by a distinction between form and case data at the pre processing step. We are just working on an xml and just checking for tags. 
I think if the pattern is written strictly, it will match something in a case create/update block or in a form meta block. The only way it would match in both is the pattern is written as such or if the same tag is used in both form and case blocks, not sure if both are even possible. However there can be two case blocks that can have the same case property.
To try and answer your question,
In case the same match happens more than once, a single vault entry is created for each "secret text" and the corresponding vault key would be used for all occurrences of the "secret text", for example in this instance if an aadhaar number appears more than once in the xml, all of its occurrences will be replaced by the same key.
Reference for change: https://github.com/dimagi/commcare-hq/pull/28161/files#diff-da01187de21bca064166b847136e5c97R41

Reply:
So in any form that's built in app manager you can't put save case properties directly, you need a question and then you can copy the value to a case property. This means you end up with something like this in the data:

```xml
<data>
  <description_group>
    <description>Test</description>
  </description_group>
  <followup_date>2013-12-17</followup_date>
  <meta>...</meta>
  <case ...>
    <update>
      <date_followup>2013-12-17</date_followup>
      <description>Test</description>
    </update>
  </case>
</data>
```

In some cases the question data element will be named the same as the case property but that is at the discretion of the app builder. 

Either way I think you answered the question, provided the regex patterns match it should work and won't create duplicate vault entries.